File: course1_algorithmic_toolbox/week3_greedy_algorithms/7_maximum_salary/largest_number.js
const readline = require('readline');
const rl = readline.createInterface({
	input: process.stdin,
	terminal: false
});

process.stdin.setEncoding('utf8');

const isGreaterOrEqual = (x, y) => {
	return parseInt(`${x}${y}`, 10) >= parseInt(`${y}${x}`, 10);
};

const swap = (a, i, j) => {
	const tmp = a[i];
	a[i] = a[j];
	a[j] = tmp;
};

const largestNumber = (a) => {
	for (let i = 0; i < a.length - 1; i++) {
		for (let j = i + 1; j < a.length; j++) {
			if (isGreaterOrEqual(a[j], a[i])) {
				swap(a, i, j);
			}
		}
	}

	return a.join('');
};

/**
 * Enter:
 *
 * 2
 * 21 2
 *
 */

rl.once('line', line => {
	rl.once('line', line => {
		const nums = line.toString().split(' ').map(Number);
		console.log(largestNumber(nums));
		process.exit();
	});

});


module.exports = largestNumber;

File: course1_algorithmic_toolbox/week3_greedy_algorithms/7_maximum_salary/LargestNumber.java
import java.util.Scanner;

public class LargestNumber {

	private static String largestNumber(String[] a) {
		for (int i = 0; i < a.length - 1; i++) {
			for (int j = i + 1; j < a.length; j++) {
				if (isGreaterOrEqual(a[j], a[i])) {
					swap(a, i, j);
				}
			}
		}

		StringBuilder num = new StringBuilder();
		for (String s : a) {
			num.append(s);
		}

		return num.toString();
	}

	private static void swap(String[] a, int i, int j) {
		String tmp = a[i];
		a[i] = a[j];
		a[j] = tmp;
	}

	private static boolean isGreaterOrEqual(String x, String y) {
		return Integer.parseInt(x + y) >= Integer.parseInt(y + x);
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		String[] a = new String[n];
		for (int i = 0; i < n; i++) {
			a[i] = scanner.next();
		}
		System.out.println(largestNumber(a));
	}
}

